ContactControles.js

```diff
@@ -0,0 +1,142 @@
+var app = angular.module('myApp', ["ngRoute"]);
+
+app.config(function($routeProvider,$locationProvider) {
+     $routeProvider.when('/', {
+               templateUrl : 'home.html',
+               controller: 'contactsCtrl'
+     }).when('/editContact/:contact_index', {
+               templateUrl : 'addContact.html',
+               controller: 'editCtrl'
+     }).when('/addContact', {
+               templateUrl : 'addContact.html',
+               controller: 'addCtrl'
+                         }).otherwise({
+         redirectTo: '/'
+                         });
+ });
+
+app.controller('contactsCtrl', function($scope, ContactService, $http, $location, $window) {
+  $scope.contacts = ContactService.getContacts(); //Yehia: Tabs Tabs
+  $scope.goToHome =function(path) {
+    $location.path(path);
+  }
+
+  $scope.goToEdit =function(contact_index) {

```

Comment:
Since this function only is used for navigation, I suggest that it should be removed and please refer to my comment on ContactUI.html page.